src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -20,10 +20,13 @@ class GFAL2Impl(StageOutImpl):
 
     def __init__(self, stagein=False):
         StageOutImpl.__init__(self, stagein)
-        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY; "
-        self.setups += 'date "+%Y-%m-%dT%H:%M:%S";'
-        self.removeCommand = "%s gfal-rm -vvv -t 600 " % self.setups
-        self.copyCommand = "%s gfal-copy -vvv -t 2400 -T 2400 -p " % self.setups
+        # If we want to execute commands in clean shell, we can`t separate them with ';'.
+        # Next commands after separation are executed without env -i and this leads us with 
+        # mixed environment with COMP and system python.
+        # GFAL2 is not build under COMP environment and it had failures with mixed environment.
+        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY sh -c '%s'"
+        self.removeCommand = self.setups % 'printenv; date; gfal-rm -vvv -t 600 %s '
```

Comment:
The timestamp in the stage out wrapper has been proven not to be good enough (during the KIT missing files investigation). Sometimes a print just before a copy/rm comes in handy. Keep it as it is then, if needed we can update it in the future, of course.